### PR TITLE
GCW-3281 Add Charity Position

### PR DESCRIPTION
**Pull request by bharat619**
### Ticket Link: 
https://jira.crossroads.org.hk/browse/GCW-3281

### What does this PR do?
Add minor API support and validations for adding charity positions

### CODE REVIEW

File: app/controllers/api/v1/organisations_users_controller.rb [hunk 1]
```diff
@@ -47,6 +48,10 @@ def show
         render json: @organisations_user, serializer: serializer
       end
 
+      def user_status
+        render json: { status: OrganisationsUser.all_status }, status: 200
+      end
+
```
Comment (patrixr) [inline app/controllers/api/v1/organisations_users_controller.rb:54]: Isn't it a bit overkill to create an endpoint just for a few strings ? Maybe using a `goodcity_setting` record is more appropriate ?

You could use an initializer to create it perhaps ?

File: app/models/organisations_user.rb [hunk 2]
```diff
@@ -1,19 +1,37 @@
+# frozen_string_literal: true
+
 class OrganisationsUser < ActiveRecord::Base
   module Status
-    PENDING   = 'pending'.freeze
-    APPROVED  = 'approved'.freeze
-    EXPIRED   = 'expired'.freeze
-    DENIED    = 'denied'.freeze
```
Comment (patrixr) [inline app/models/organisations_user.rb:6]: Any reason for removing the `freeze` ?

**Review by patrixr — COMMENTED** [review]
PR looks pretty clean, I'm not so sure about the best way to share that status list with the frontend.

Typically we would store that list in a constants file in the app, although I can't find a good argument against fetching it from the backend.

File: app/controllers/api/v1/organisations_users_controller.rb (hunk 1 above)
Comment (patrixr) [reply app/controllers/api/v1/organisations_users_controller.rb:54]: I'm just thinking out loud. Any thoughts @steveyken  ?

File: app/models/organisations_user.rb (hunk 2 above)
Comment (bharat619) [reply app/models/organisations_user.rb:6]: I have added a magic comment at top `# frozen_string_literal: true` 
this handles the `.freeze` of strings : )

**Review by patrixr — APPROVED** [review]

**Review by shreyas098 — APPROVED** [review]

**Pull request merged**